### PR TITLE
Included the creation of new recipe with ingredients

### LINKED ISSUE #17: As a user, I can create new recipes so that I can keep a list of recipes in the application.

### PR DESCRIPTION
**Pull request by alwayshuizhen**
The following feature has been added:
- Adding a new recipe to the application

The following can be noticed when running the program:
- The linkage between ingredients and recipe is not yet implemented.

This pull request fixes #17

### CODE REVIEW

File: src/main/java/seedu/kitchenhelper/KitchenHelper.java [hunk 1]
```diff
@@ -13,48 +14,58 @@
...
     }
     
     private void run() {
         start();
-        runCommandLoopUntilExitCommand();
+        try {
+            runCommandLoopUntilExitCommand();
+        } catch (KitchenHelperException e) {
+            ui.printInvalidCmd();
+        }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/KitchenHelper.java:36]: Do you think this violates the Slap? 
Perhaps shifting inside runCommandLoopUntilExitCommand might be better

File: src/main/java/seedu/kitchenhelper/command/Command.java [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
+            feedbackToUser = addRecipe(objectVariables, recipeList);
+        } else if (actionType.equals(AddCommand.COMMAND_WORD) && objectType.equals("ingredient")) {
+            // change here for your part!
+            System.out.println("hello add ingr");
+        } else if (actionType.equals(AddCommand.COMMAND_WORD) && objectType.equals("chore")) {
+            System.out.println("hello add chore");
+        }
+        CommandResult cmdResult = new CommandResult(feedbackToUser);
+        return cmdResult;
     }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/Command.java:49]: Perhaps can have a class that specifically do the particular command. For single responsibility principle, under week 9 topics. 
Sorry, for the misunderstanding cause, when I help out with the base skeleton code. 
Can review AB2 or personsbook for more clarifications of the coding style implemented.

File: src/main/java/seedu/kitchenhelper/parser/Parser.java [hunk 3]
```diff
@@ -38,6 +47,32 @@ public Command parseUserCommand(String input) {
...
+                String[] nameAndType = new String[2];
+                nameAndType[0] = ingrContent[0];
+                nameAndType[1] = ingrContent[2];
+                ingrAndQty.put(nameAndType, Integer.parseInt(ingrContent[1]));
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new KitchenHelperException("Invalid Command");
+        }
+        return ingrAndQty;
+    }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/parser/Parser.java:75]: Good job of parsing the user input string.

File: src/main/java/seedu/kitchenhelper/KitchenHelper.java [hunk 4]
```diff
@@ -13,48 +14,58 @@
...
-    public static void main(String[] args) {
+    public static void main(String[] args) throws KitchenHelperException {
         new KitchenHelper().run();
     }
     
     private CommandResult executeCommand(Command command) {
         try {
-            CommandResult result = command.execute();
+            // to check if you get the right object
+            // System.out.println(command.getClass().getName());
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/KitchenHelper.java:67]: Perhaps removing these dead code or comments, can consider using JUnit5 to cover these cases.

File: src/main/java/seedu/kitchenhelper/command/AddCommand.java [hunk 5]
```diff
@@ -1,23 +1,79 @@
...
-    public void addRecipe(String attributes) {
-    
+
+    @Override
+    public String addRecipe(String attributes, ArrayList<Recipe> recipeList) throws KitchenHelperException {
+        Recipe freshRecipe = new Recipe();
+        freshRecipe.setRecipeName(attributes);
+        freshRecipe.addIngredientsToRecipe(parsedIngr);
+        return freshRecipe.recipeName + " Recipe has been created with "
+                + freshRecipe.recipeIngrQty + " ingredients inside.";
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/AddCommand.java:67]: Perhaps can consider using String.format, to reduce string literals

File: src/main/java/seedu/kitchenhelper/object/Recipe.java [hunk 6]
```diff
@@ -1,5 +1,74 @@
...
+    public void addIngredientsToRecipe(HashMap<String[], Integer> ingrAndQty) {
+        for (Map.Entry<String[], Integer> entry : ingrAndQty.entrySet()) {
+            String ingrName = (entry.getKey())[0];
+            String ingrCategory = (entry.getKey())[1];
+            Integer ingrQuantity = entry.getValue();
+            Ingredient newIngredient = createIngr(ingrName, ingrCategory, ingrQuantity);
+            recipeItems.add(newIngredient);
+        }
+        recipeIngrQty = recipeItems.size();
+    }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/object/Recipe.java:39]: Good work of having methods related to recipe in the Recipe class 👍

File: src/main/java/seedu/kitchenhelper/object/Recipe.java [hunk 7]
```diff
@@ -1,5 +1,74 @@
...
+        case "miscellaneous":
+            return new Miscellaneous(ingrName, ingrCategory, ingrQuantity, 0, null);
+        case "staple":
+            return new Staple(ingrName, ingrCategory, ingrQuantity, 0, null);
+        case "vegetable":
+            return new Vegetable(ingrName, ingrCategory, ingrQuantity, 0, null);
+        default:
+            return null;
+        }
+    }
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/object/Recipe.java:68]: Good work of having methods related to recipe in the Recipe class 👍

File: src/main/java/seedu/kitchenhelper/parser/Parser.java [hunk 8]
```diff
@@ -18,14 +25,16 @@
...
         final String commandWord = userInputs[0];
         final String parameters = userInputs[1];
-        
         switch (commandWord.toLowerCase()) {
         case AddCommand.COMMAND_WORD:
-            return new AddCommand();
+            AddCommand addCmd = new AddCommand();
+            HashMap<String[], Integer> ingrAndQty = prepareAddRecipe(parameters);
+            addCmd.setAttributesOfCmd(parameters, ingrAndQty);
+            return addCmd;
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/parser/Parser.java:37]: Consider implementing Slap in this part of the code

**Review by jinfayap — COMMENTED** [review]
Here are the changes that could be made. If there are any inquiry, do let me know.

File: src/main/java/seedu/kitchenhelper/KitchenHelper.java (hunk 1 above)
Comment (alwayshuizhen) [reply src/main/java/seedu/kitchenhelper/KitchenHelper.java:36]: I don't think it is. An abstraction is when you take a complex subject, or entity and hide most of its details while exposing the functionality that still defines the essence of that object.

File: src/main/java/seedu/kitchenhelper/KitchenHelper.java (hunk 4 above)
Comment (alwayshuizhen) [reply src/main/java/seedu/kitchenhelper/KitchenHelper.java:67]: This was meant to be a _guide_ for the team to check if they have managed to return the object type that they wanted. It is not a dead code/ comment.

File: src/main/java/seedu/kitchenhelper/command/AddCommand.java (hunk 5 above)
Comment (alwayshuizhen) [reply src/main/java/seedu/kitchenhelper/command/AddCommand.java:67]: It doesn't matter I guess?

File: src/main/java/seedu/kitchenhelper/command/Command.java (hunk 2 above)
Comment (alwayshuizhen) [reply src/main/java/seedu/kitchenhelper/command/Command.java:49]: I think we can use your idea after its accepted and agreed by the team at our regular meetings. I feel that it should not be highlighted only during pull request and this is not something that will only affect the functions that I have but to everyone. I can always edit this again after we agree to it 😄

Comment (jinfayap) [reply src/main/java/seedu/kitchenhelper/command/Command.java:49]: I will merge your PR, and leave it as it is. Perhaps, I will do up another way of implementation, if that doesn't work out, then i change back to the team implementation.

**Pull request merged**